src/track/steminfoimporter.cpp

```diff
@@ -0,0 +1,144 @@
+#include "track/steminfoimporter.h"
+
+#include <QFile>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QString>
+
+#include "util/logger.h"
+
+#define ATOM_TYPE(value)                                                   \
+    (uint32_t)(((uint8_t)(value)[0] << 24) | ((uint8_t)(value)[1] << 16) | \
+            ((uint8_t)(value)[2] << 8) | (uint8_t)(value)[3])
+
+namespace mixxx {
+
+namespace {
+
+const mixxx::Logger kLogger("StemInfoImporter");
+constexpr int kSupportedStemVersion = 1;
+const QStringList kStemFileExtensions = {".stem.mp4", ".stem.m4a"};
+
+const uint32_t kAtomHeaderSize = 8; // 4 bytes (unsigned integer) + 4 char
+const uint32_t kStemManifestAtomPath[] = {
+        ATOM_TYPE("moov"),
+        ATOM_TYPE("udta"),
+        ATOM_TYPE("stem"),
+        0 // This indicate end of the path
+};
+
+/// @brief Seek the reader the atom requested.
+/// @param reader The IODevice to search MP4 atom in
+/// @param path The list of atom to traverse in the tree, from top to bottom.
+/// Must be null terminated
+/// @param boxSize The size of the box currently under the cursor to focus the
+/// search in
+/// @param pathIdx The level of the tree currently search (index of path)
+/// @return the size of the box data if found, -1 otherwise
+uint32_t seekTillAtom(QIODevice& reader,
+        const uint32_t path[],
+        uint32_t boxSize = 0,
+        uint32_t pathIdx = 0) {
+    if (!path[pathIdx]) {
+        return boxSize;
+    }
+
+    uint32_t byteRead = 0;
+    char buffer[kAtomHeaderSize];
+    while (!reader.atEnd() || (boxSize && byteRead >= boxSize)) {
+        reader.read(buffer, kAtomHeaderSize);
+        byteRead += boxSize;
+        if (ATOM_TYPE(buffer + 4) == path[pathIdx]) {
+            return seekTillAtom(reader, path, ATOM_TYPE(buffer) - kAtomHeaderSize, pathIdx + 1);
+        }
+        reader.skip(ATOM_TYPE(buffer) - kAtomHeaderSize);
+    }
+    return -1;
+}
+
+} // anonymous namespace
+
+// static
+bool StemInfoImporter::isStemFile(
```

Comment:
Here we need apply the mime type logic. And probably the checking for easy "stem" properties in addition.
Alternative  this may look for these properties only and we can call the function isMp4AStem() or such.  
This raises the general question if this Class can be only used with MP4s or if we need to protect every invocation to be called with other files.

Reply:
> Here we need apply the mime type logic

AFAIK, mime type isn't available in the Track object. Reading the file (to detect the MIME using `QMimeDatabase::mimeTypeForFile` will likely have effect on performance. of every single file being imported, which is likely not negligible when importing a large library. 

> This raises the general question if this Class can be only used with MP4s

I'm not sure why we would want to support another file type. The STEM atom is part of the NI standard, meaning it needs to be in a mp4 file. If we want to implement other STEM standards, there is no guarantee that there will use a STEM atom, as part of the MPEG standard. My suggestion would be to keep the implantation to what we know and support currently (NI standard, mp4) and cross the bridge to support other standards when we decide to do so.

Reply:
I think that that detection based on `stem.mp4` extension is overly restrictive. At the extreme any iso media container could be feature tested for `moov.udta.stem`, but at the least we should be able to detect stems in arbitrary `.mp4` containers.

Reply:
Again, I was simply implementing the specification (which I believe @dszakallas also had access to). Happy to go beyond what it defines.

Reply:
We should not expand this PR by general MP4 support. Scope of this PR is the STEMs format and nothing beyond!
Traktor Pro seems to have another detection logic than just checking for .stems.mp4 extension, because it seems to work also with the extension .stems.m4a (or they just check for one of both file extensions).
Performance for file type detection is important in my opinion. What would it cost to probe for moov.udta.stem ?

Reply:
We already support mp4 some way, as I was able to drop mp4 tracks into mixxx and play them. And the specification says that the container should be an ISOBMFF with `.mp4` as extension. `.stem.mp4` is "preferred", not required. So, if we want to follow the spec we should try to detect stems in any mp4.

Reply:
> What would it cost to probe for moov.udta.stem ?

That's hard to assess as the atom offset isn't static and require to traverse boxes in the file. 

> extension. .stem.mp4 is "preferred", not required

That's true but in the meantime, Mixxx aims to be vendor-agonistic IMO, so the flexibility that Tracktor Pro gives to arrange NI shouldn't drive Mixxx implementation. Anyway, last commit adds support for `.mp4`, but happy to remove depending where the consensus goes.